src/types/full_block.py

```diff
@@ -170,3 +174,13 @@ def additions_for_npc(npc_list: List[NPC]) -> List[Coin]:
             additions.append(coin)
 
     return additions
+
+
+def announcements_for_npc(npc_list: List[NPC]) -> List[Announcement]:
+    announcements: List[Announcement] = []
+
+    for npc in npc_list:
+        for coin in created_announcements_for_conditions_dict(npc.condition_dict, npc.coin_name):
```

Comment:
I think you can use `announcements.extend(created_announcements_for_conditions_dict(npc.condition_dict, npc.coin_name)` here

Reply:
Then you don't need the inner loop.

Reply:
implemented